src/Configure/Config.php

```diff
@@ -0,0 +1,163 @@
+<?php
+namespace RabbitMQ\Configure;
+
+use Cake\Core\Configure;
+
+/**
+ * Class for loading the configuration and check any error
+ */
+class Config
+{
+    // Configuration key prefix
+    const CONFIGURE_KEY = 'Riesenia.CakephpRabbitMQ';
+    const CONFIGURE_KEY_PREFIX = self::CONFIGURE_KEY . '.';
+    const SERVER_CONFIGURE_KEY = self::CONFIGURE_KEY_PREFIX . 'server';
+
+    // Default configuration file
+    const DEFAULT_CONFIG = __DIR__ . '/../../config/default_config.php';
+    const SERVER_DEFAULT = __DIR__ . '/../../config/default_server.php';
+
+    // Supported callback type
+    const CALLBACK_TYPE = ['callback', 'command', 'cake_command'];
+
+    // Configuration Cache
+    protected static $_configs = [];
+    protected static $_server = [];
+
+    /**
+     * Get all queue configurations (execpt server config)
+     *
+     * @return array all queue configurations
+     */
+    public static function getAllConfigs()
+    {
+        $userConfigs = Configure::read(self::CONFIGURE_KEY);
+        unset($userConfigs['server']);
+        return static::getConfigs(array_keys($userConfigs));
+    }
+
+    /**
+     * Get queue configurations according to the keys provided
+     *
+     * @param  array $keys
+     * @return array queue configurations
+     */
+    public static function getConfigs(array $keys)
+    {
+        $configs = [];
+        foreach ($keys as $key) {
+            $configs[$key] = static::get($key);
+        }
+        return $configs;
+    }
+
+    /**
+     * Get a queue configuration according to the key
+     *
+     * @param  string $key
+     * @return array queue configuration
+     * @throws InvalidArgumentException if config specified by key not found\
+     * @throws InvalidArgumentException if key is 'server'
+     * @throws InvalidArgumentException if non or more than one callback type is provided
+     */
+    public static function get(string $key)
+    {
+        if (isset(static::$_configs[$key])) {
+            return static::$_configs[$key];
+        }
+
+        // Check if the config required exist
+        if (!Configure::check(self::CONFIGURE_KEY_PREFIX . $key)) {
+            throw new \InvalidArgumentException('Queue "' . $key . '" is not configured');
+        }
+
+        // Check if the key is not server
+        if ($key == 'server') {
+            throw new \InvalidArgumentException('Server configuration should not get by ' . __FUNCTION__ . ' function');
+        }
+
+        // Check one and only one callback is provided
```

Comment:
ok. i see you are validating callbacks here. so what's the point in separating the logic?